packages/roosterjs-content-model-core/lib/coreApi/restoreUndoSnapshot.ts

```diff
@@ -23,8 +24,28 @@ export const restoreUndoSnapshot: RestoreUndoSnapshot = (core, snapshot) => {
     try {
         core.undo.isRestoring = true;
 
+        // reset logical root to physical root
+        core.api.setLogicalRoot(core, null);
```

Comment:
I'm reconsidering about this, maybe no need to set to root here, but always set to correct logical root below when restoreLogicalRoot.

For example:

```ts
function restoreLogicalRoot() {
   const newLogicalRoot = snapshot.logicalRootPath?.length > 0 ? getNodeFromPath(snapshot.logicalRootPath) : core.physicalRoot;
   setLogicalRoot(newLogicalRoot);
}
```